js_src/vite.config.mjs

```diff
@@ -0,0 +1,42 @@
+import { resolve, dirname, parse } from 'path';
+import { defineConfig } from 'vite';
+import pkg from './package.json' assert { type: 'json' };
+
+const input = 'src/index.js';
+const jsPackageName = 'rtl';
+
+const globals = {
+  react: 'React',
+  'react-dom': 'ReactDOM',
+};
+
+export default defineConfig({
+  define: {
+    'process.env.NODE_ENV': JSON.stringify('production'),
```

Comment:
Hmm I never noticed this, but I would think for this package that we'd actually want this to be `'development'`... Since we'd want more assertions in a test environment, right?

Wonder what rollup was doing before. Or, if it really makes much of a difference; looks like there's not a lot of checking of these at least in RTL itself: https://github.com/search?q=repo%3Atesting-library%2Freact-testing-library%20NODE_ENV&type=code

So we're probably good to just switch it?